refiner-lib/src/refiner_inner.rs

```diff
@@ -543,19 +547,28 @@ fn build_transaction(
                 }
                 InnerTransactionKind::Call => {
                     hash = virtual_receipt_id.0.try_into().unwrap();
-                    tx = tx
-                        .hash(hash)
-                        .from(near_account_to_evm_address(predecessor_id.as_bytes()));
+                    let from_address = near_account_to_evm_address(predecessor_id.as_bytes());
+
+                    tx = tx.hash(hash).from(from_address);
 
                     if let Some(call_args) = CallArgs::deserialize(&bytes) {
                         let (to_address, value, input) = match call_args {
                             CallArgs::V2(args) => (args.contract, args.value, args.input),
                             CallArgs::V1(args) => (args.contract, WeiU256::default(), args.input),
                         };
 
+                        let nonce = storage
+                            .with_engine_access(
+                                near_block.header.height,
+                                transaction_index.try_into().unwrap(),
```

Comment:
Forgot to ask you if the `transaction_index` is the right value for the `transaction_position` argument.

Reply:
Let me know if this is being tested by the general flow of the test or if we should add some assertion.